lib/engine/ability/base.rb

```diff
@@ -9,14 +9,17 @@ class Base
       include Helper::Type
       include Ownable
 
+      attr_accessor :name, :value
       attr_reader :type, :owner_type, :remove, :when, :count
 
-      def initialize(type:, owner_type: nil, count: nil, remove: nil, **opts)
+      def initialize(type:, name: nil, owner_type: nil, count: nil, remove: nil, value: nil, **opts)
         @type = type&.to_sym
+        @name = name.nil? ? @type : name&.to_s
```

Comment:
@name = name&.to_s || @type

Reply:
do we really need name? can't we just use the type, you've copied and pasted historical object multiple times, destination may be fine

Reply:
Applied 1st suggestion in next push.
Name field is used when presenting the ability on the corporation charter.